File: cpp/que/main_10866.cpp
#include <iostream>
#include <deque>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);

	int count;
	cin >> count;
	deque<int> deq;
	for (int i = 0; i < count; ++i) {
		string command;
		cin >> command;

		int num;
		if (command == "push_front") {
			cin >> num;
			deq.push_front(num);
		} else if (command == "push_back") {
			cin >> num;
			deq.push_back(num);
		} else if (command == "pop_front") {
			if (deq.empty()) {
				cout << -1 << '\n';
			} else {
				cout << deq.front() << '\n';
				deq.pop_front();
			}
		} else if (command == "pop_back") {
			if (deq.empty()) {
				cout << -1 << '\n';
			} else {
				cout << deq.back() << '\n';
				deq.pop_back();
			}
		} else if (command == "size") {
			cout << deq.size() << '\n';
		} else if (command == "empty") {
			cout << deq.empty() << '\n';
		} else if (command == "front") {
			if (deq.empty()) {
				cout << -1 << '\n';
			} else {
				cout << deq.front() << '\n';
			}
		} else if (command == "back") {
			if (deq.empty()) {
				cout << -1 << '\n';
			} else {
				cout << deq.back() << '\n';
			}
		}
	}
}

File: kotlin/step/que/main_10866.kt
fun main() {
    val count = readLine()!!.toInt()
    val deque = ArrayDeque<Int>(count)
    for (i in 1..count){
        val command = readLine()!!.split(' ')
        when(command[0]){
            "push_front" -> deque.addFirst(command[1].toInt())
            "push_back" -> deque.add(command[1].toInt())
            "pop_front" -> println(deque.removeFirstOrNull() ?: -1)
            "pop_back" -> println(deque.removeLastOrNull() ?: -1)
            "size" -> println(deque.size)
            "empty" -> println( if (deque.isEmpty()) 1 else 0 )
            "front" -> println(deque.firstOrNull() ?: -1)
            "back" -> println(deque.lastOrNull() ?: -1)
        }
    }
}